scalding-core/src/main/scala/com/twitter/scalding/FileSource.scala

```diff
@@ -371,30 +367,21 @@ trait SequenceFileScheme extends SchemedSource {
 }
 
 /**
- * Ensures that a _SUCCESS file is present in every directory included by a glob,
- * as well as the requirements of [[FileSource.pathIsGood]]. The set of directories to check for
- * _SUCCESS
- * is determined by examining the list of all paths returned by globPaths and adding parent
- * directories of the non-hidden files encountered.
- * pathIsGood should still be considered just a best-effort test. As an illustration the following
- * layout with an in-flight job is accepted for the glob dir*&#47;*:
- * <pre>
- *   dir1/_temporary
- *   dir2/file1
- *   dir2/_SUCCESS
- * </pre>
+ * Uses _SUCCESS files instead of the presence of non-hidden files to determine if a path is good.
  *
- * Similarly if dir1 is physically empty pathIsGood is still true for dir*&#47;* above
+ * Requires that:
+ * 1) Every matched, non-hidden directory contains a _SUCCESS file
+ * 2) Every matched, non-hidden file's parent directory contain a _SUCCESS file
  *
- * On the other hand it will reject an empty output directory of a finished job:
- * <pre>
- *   dir1/_SUCCESS
- * </pre>
+ * pathIsGood should still be considered just a best-effort test. There are still cases where this is
+ * not a sufficient test for correctness. See <ticket>
```

Comment:
what is the `<ticket>` Can we add some examples of how this is broken and even warn people off of the whole design?

Ideally, we want this sanity checking in the mappers, I think. And we fail the tasks if they can't read the data. But this is getting at atomicity in jobs, and that will depend on the underlying fs (we use s3 a lot at Stripe, as I imagine many folks do), so the issue is quite complex. Maybe worth really writing a doc on?